clients/fides-js/src/lib/consent-banner.tsx

```diff
@@ -0,0 +1,122 @@
+import { h, Component, VNode} from 'preact';
+import { useState } from 'preact/hooks';
+import {CookieKeyConsent, hasSavedConsentCookie, setConsentCookieAcceptAll, setConsentCookieRejectAll,} from "./cookie";
+import {ConsentBannerOptions} from "~/lib/consent-types";
+import debugLog from "~/lib/consent-utils";
+
+
+
+interface BannerProps {
+    defaults: CookieKeyConsent;
+    options: ConsentBannerOptions;
+}
+
+interface BannerState {
+    isShown: boolean
+}
+
+class ConsentBanner extends Component<BannerProps, BannerState> {
+
+    /**
+     * Builds a button DOM element with the given id, class name, and text label
+     */
+    static buildButton = (id: string, className: string, label?: string, onClick?: () => void) => {
+        return(<button id={id} class={`fides-consent-banner-button ${className}`} onClick={onClick ? onClick : undefined}>
+            {label || ""}
+        </button>);
+    }
+
+    /**
+     * Navigates to the Fides Privacy Center to manage consent settings
+     */
+    navigateToPrivacyCenter = (): void => {
+        debugLog("Navigate to Privacy Center URL:", this.options.privacyCenterUrl);
+        if (this.options.privacyCenterUrl) {
+            window.location.assign(this.options.privacyCenterUrl);
+        }
+    }
+
+
+    private readonly defaults: CookieKeyConsent;
+
+    private readonly options: ConsentBannerOptions;
+
+    constructor(props: BannerProps) {
+        super(props);
+        this.defaults = props.defaults;
+        this.options = props.options;
+    }
+
+    /**
+     * Builds the DOM elements for the consent banner (container, buttons, etc.) and
+     * return a single div that can be added to the body.
+     */
+    buildBanner = (defaults: CookieKeyConsent) => {
+        const [isShown, setisShown] = useState(false);
+        // TODO: support option to specify top/bottom
+
+        // Show the banner after a small delay, to allow animation to occur
+        setTimeout(() => setisShown(true), 100);
+        return (
+            <div id="fides-consent-banner" class={`fides-consent-banner fides-consent-banner-bottom ${isShown ? "" : "fides-consent-banner-hidden"} `}>
+                <div id="fides-consent-banner-description" class="fides-consent-banner-description">
+                    {this.options.labels?.bannerDescription || ""}
+                </div>
+                <div id="fides-consent-banner-buttons" class="fides-consent-banner-buttons">
+                    {ConsentBanner.buildButton(
+                        "fides-consent-banner-button-tertiary",
+                        "fides-consent-banner-button-tertiary",
+                        this.options.labels?.tertiaryButton,
+                        this.navigateToPrivacyCenter,
+                    )}
```

Comment:
This should be something like `<ConsentBannerButton type="tertiary" onClick={navigateToPrivacyCenter} />`